File: Mastering Unity 2D/Assets/Scripts/Entity.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Entity : ScriptableObject
{
    public string name_;
    public int age_;
    string faction_;
    public string occupation_;
    public int level_ = 1;
    public int health_ = 2;
    public int strength_ = 1;
    public int magic_ = 0;
    public int defense_ = 0;
    public int speed_ = 1;
    public int damage_ = 1;
    public int armor_ = 0;
    public int no_Of_Attacks_ = 1;
    public string weapon_;
    public Vector2 position_;

    public void Take_Damage(int amount)
    {
        health_ -= Mathf.Clamp((amount - armor_), 0, int.MaxValue);
    }

    public void Attack(Entity other)
    {
        other.Take_Damage(strength_);
    }

    
}

File: Wyvern/Wyvern/source/Entity.cpp
// Entity.cpp

#include "Entity.h"

void Entity::Create()
{

}

void Entity::Create(Vector2 position, Vector2 size)
{
	position_ = position;
	size_ = size;
}

void Entity::Create(Vector2 position, float w, float h)
{
	position_ = position;
	size_ = Vector2(w, h);
}

void Entity::Create(float x, float y, Vector2 size)
{
	position_ = Vector2(x, y);
	size_ = size;
}

void Entity::Create(float x, float y, float w, float h)
{
	position_ = Vector2(x, y);
	size_ = Vector2(w, h);
}

void Entity::Create(int x, int y, int w, int h)
{
	position_ = Vector2(x, y);
	size_ = Vector2(w, h);
}

